src/main/java/com/minecolonies/coremod/entity/pathfinding/PathJobWalkRandomEdge.java

```diff
@@ -0,0 +1,68 @@
+package com.minecolonies.coremod.entity.pathfinding;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Walks to a random edge block nearby, a block next to air. Does not use ladders
+ */
+public class PathJobWalkRandomEdge extends AbstractPathJob
+{
+    private static final Direction[] directions = {Direction.NORTH, Direction.SOUTH, Direction.EAST, Direction.WEST};
```

Comment:
There is already a horizontal directions array in vanilla you can query afaik

Reply:
hm only seeing a private one

Reply:
Direction.Plane.HORIZONTAL